### ISSUE TITLE
Freezing and clipping when transform-ing large images

### ISSUE THREAD

**Issue by marchare555** [renpy/renpy#3965]
Since updating to 7.5.3, I seem to be experiencing a regression where a large image (layered or not) freezes when transform-ed.

Here's the code where it happens:

```
   show picture with dissolve:
        zoom 1 xalign 0.65 yalign 1.0
        easein 4.0 align (0.65, 0.4) knot (0.5, 0.8) knot (0.8, .6) knot (0.6, 0.5)
```

If I remove the knots, "picture" instead takes a good 2-3 seconds to load but then transforms from bottom to top without clipping.

I was previously using 7.4.11, where it worked properly with no clipping or freezing. I am now using 7.5.3.22090602.

**Comment by renpytom** [comment]
How large is large? What is "picture"?

**Comment by marchare555** [comment]
"picture" is a 3840x2160 CG, for a 1920x1080 game window resolution.

**Comment by renpytom** [comment]
Can you send a short example game? My guess is that if your picture is taking seconds to load, that's probably an issue with the picture rather than Ren'Py.

**Comment by marchare555** [comment]
So, after checking the image load log, it turns out that this particular moment was clocking +95% of the default, 300 MB image cache. This also happens if I replace "picture" with a small, 500x500 png image.

If I up the cache size to 400 MB, everything gets loaded with no lag. The cache gets filled to ~72%.

Now, if I load the same scene with Ren'py 7.4.11 and a 300 cache size, the same scene gets to something like ~75% of the cache, and never goes more than 81%. I'm not sure why 7.5.3 is using more cache than 7.4.11

And I'm afraid I can't sent the images in question; they're NDA, sorry.

**Comment by renpytom** [comment]
The NDA is the choice of your project. 

I need an example I can work with to be able to diagnose and solve what's wrong.

**Comment by marchare555** [comment]
I tried doing it in another project with other image assets, but couldn't reproduce the bug. :(  Both CG #1 and CG #2 are 4K images comprised of a dozen of transparent layers, and both are in 4K resolution; reproducing the same thing with newer assets would be complicated.

Another difference I noticed is that there is a difference in how 7.4.11 and 7.5.3 predict upcoming images.

In 7.5.3, If I show CG #1, then CG #2 ("picture"), then back to CG #1 again during a scene, then Ren'py will try and fail to predict CG#1 at the exact same time CG #2 is shown.

In 7.4.11, Ren'py doesn't try predicting right when CG #2 is shown, but a bit later during another "show" statement, where a lag does occur with the same failed attempts at loading CG #1's layers.

So, rather than saying my issue isn't happening in 7.4.11, it's more like it occurs later, rather than not happening at all.

Anyway, that issue seems rather specific to the project I am working on, and I am fine with the workaround by upping the cache limit. If you still want to investigate the issue further, I can try finding a better way to reproduce it; but if not, then this issue can be closed.

Thanks for your time.
